### ISSUE TITLE
Could wallet new command support no-password?

### ISSUE THREAD

**Issue by lispking** [FuelLabs/forc-wallet#202]
Could you consider supporting the "no-password" parameter when executing "forc wallet new", so that it is convenient for scripts to directly generate wallets without a password?

https://github.com/FuelLabs/fuelup/issues/660

**Comment by sdankel** [comment]
Hi @lispking , could you explain more about your use case? We require wallets to have a password as a security measure.

**Comment by kayagokalp** [comment]
I don't think we should have the option to create wallets without password. But for improving usability inside interactive context, we can accept them as a parameter, with a fair warning attached to the related doc string. This param should then be available for creating and also using the wallet, anywhere we currently ask for a password. That would be my advice but curious about what @FuelLabs/tooling thinks about this

**Comment by lispking** [comment]
To be precise, it's not about not using a password, but rather about including the password directly in the command line when creating a wallet. This makes it convenient to create a wallet with a single command in the command line, without the need for multiple interactions to create the wallet. @sdankel @kayagokalp 

like this: https://github.com/FuelLabs/forc-wallet/pull/201/files

**Comment by lispking** [comment]
> Hi @lispking , could you explain more about your use case? We require wallets to have a password as a security measure.

The use case is as follows: recently, I've been integrating forc tool into our service and found that this kind of command interactivity is not very convenient for the expansion of our service. 

This includes the subsequent `wallet import` command, which also needs to support the usage of `private_key + password` in the command line. This way, users can deploy their own `Sway` contracts with a single command using their own wallets. 

For more information, please refer to the website: https://docs.amphitheatre.app/web3-developer-guide/blockchains/

**Comment by sdankel** [comment]
It looks like the [foundry wallet](https://book.getfoundry.sh/reference/cast/cast-wallet-new) CLI has an `--unsafe-password` option, and also lets you set a password via environment variable. We could go this route.

**Comment by lispking** [comment]
> It looks like the [foundry wallet](https://book.getfoundry.sh/reference/cast/cast-wallet-new) CLI has an `--unsafe-password` option, and also lets you set a password via environment variable. We could go this route.
> 
> The CLI argument really isn't safe an shouldn't be used in any production code other than tests. Perhaps we should only allow the env variable option. Would that satisfy your use case @lispking ?

yes, you can. renamed it: https://github.com/FuelLabs/forc-wallet/pull/201/commits/60d035b458f5c6367b797e57199b282dc11a9da8

**Comment by sdankel** [comment]
To clarify, I think we should still _not_ add a CLI argument, but instead support reading the password from an environment variable like `FORC_WALLET_PASSWORD`.

**Comment by lispking** [comment]
> To clarify, I think we should still _not_ add a CLI argument, but instead support reading the password from an environment variable like `FORC_WALLET_PASSWORD`.

Yeah, Using environment variables is also an option, but the CLI tool could be extended to support configuring some environment variables, thereby giving the choice to the user.

**Comment by lispking** [comment]
Certainly, I hope that there won't be too much interactive information printed [here](https://github.com/FuelLabs/forc-wallet/pull/201/files), so my patch also adds a silent capability.

**Comment by sdankel** [comment]
> A good practice is to support both environment variables and command-line options; many tools implement it this way. @sdankel 

That's generally true but not for passwords.
